10/umbraco-deploy/deployment-workflow/import-export.md

```diff
@@ -594,4 +598,71 @@ internal class ArtifactImportExportComposer : IComposer
     }
 }
 ```
-</details>
\ No newline at end of file
+
+</details>
+
+## Upgrade your Project from Version 7
```

Comment:
Suggested update from here to the end of the file, in terms of heading, indentation and content, is as follows.  I'm removed some parts that aren't really specific to Umbraco 7, adjusted indentation so it's clearly part of the "Migrating From Umbraco 7" section and explained a bit more background as to why these steps are necessary to get hold of Deploy for Umbraco 7.

---

#### Obtaining Umbraco Deploy for Umbraco 7

Umbraco Deploy for Umbraco 7 is no longer supported, and was only available on Umbraco Cloud.  It was not released for use on-premise.  

As such if you are looking to migrate from an Umbraco Cloud project running on Umbraco 7, you already have Umbraco Deploy installed.

If you have an Umbraco 7 on-premise website, you can use this guide to migrate from on-premise to Umbraco Cloud or to upgrade to a newer Deploy version on-premise. You will need to obtain and install Umbraco Deploy for Umbraco 7 into your project, solely for the purpose of using the export feature.

The export feature can be used without a license.

{% hint style="info" %}
A license is required for the Umbraco project you are importing into - whether that's a license that comes as part of an Umbraco Cloud subscription, or an on-premise one.
{% endhint %}

Use this guide to migrate from on-premise to Umbraco Cloud or to upgrade to a newer Deploy version on-premise.

1. Download the required `.dll` files for Umbraco Deploy for Umbraco 7 from the following links:

- [Umbraco Deploy v2.1.6](https://umbraconightlies.blob.core.windows.net/umbraco-deploy-release/UmbracoDeploy.v2.1.6.zip): Latest Deploy Version 2 release for Umbraco CMS Version 7 (officially for use on Cloud)
- [Umbraco Deploy Contrib v2.0.0](https://umbraconightlies.blob.core.windows.net/umbraco-deploy-contrib-release/UmbracoDeploy.Contrib.2.0.0.zip): Latest/only Deploy Contrib Version 2
- [Umbraco Deploy Export v2.0.0](https://github.com/umbraco/Umbraco.Deploy.Contrib/releases/tag/release-2.0.0-export): For exporting all content/schema in Version 7

2. Install Umbraco Deploy with the Contrib and Export extensions.

- Install `Umbraco Deploy`, `Umbraco.Deploy.Contrib`, and `Umbraco.Deploy.Export` by copying the downloaded `.dll` files into your Umbraco 7 site.
- When copying the files over from `Umbraco Deploy` you should not overwrite the following files (if you already had Umbraco Deploy installed):

```csharp
 Config/UmbracoDeploy.config
 Config/UmbracoDeploy.Settings.config
```

- Run the project to make sure it runs without any errors

3. Update the `web.config` file with the required references for Umbraco Deploy:

{% code title="web.config" lineNumbers="true" %}

```xml
<?xml version="1.0"?> 
<configSections>
    <sectionGroup name="umbraco.deploy">
      <section name="environments" type="Umbraco.Deploy.Configuration.DeployEnvironmentsSection, Umbraco.Deploy" requirePermission="false" />
      <section name="settings" type="Umbraco.Deploy.Configuration.DeploySettingsSection, Umbraco.Deploy" requirePermission="false" />
    </sectionGroup>
  </configSections>
  <umbraco.deploy>
    <environments configSource="config\UmbracoDeploy.config" />
    <settings configSource="config\UmbracoDeploy.Settings.config" />
  </umbraco.deploy>
</configuration>
```

{% endcode %}

4. Export and Import Content.

- **Export** your content, schema, and files to zip.